packages/l10n/scripts/generate-l10n-data.js

```diff
@@ -24,6 +24,27 @@ const supportedLocales = ['en', 'de', 'es', 'fr-FR', 'zh-CN', 'ja'];
 // https://www.drupal.org/project/drupal/issues/2036219
 const excludedCountries = ['QO', 'UN', 'ZZ'];
 
+// getNorthernIrelandCountryCode returns the Northern Ireland country code (key: name) depending on the passed locale
+// "XI" is the countryCode used to identify the Northern Ireland
+const getNorthernIrelandCountryCode = (locale) => {
+  switch (locale) {
+    case 'en':
+      return { XI: 'Northern Ireland' };
+    case 'de':
+      return { XI: 'Nordirland' };
+    case 'es':
+      return { XI: 'Irlanda del Norte' };
+    case 'fr-Fr':
+      return { XI: 'Irlande du Nord' };
+    case 'zh-CN':
+      return { XI: '北爱尔兰' };
+    case 'ja':
```

Comment:
Only challenge is not forgetting to add a locale here given we introduce another one but it's not avoidable until it's fixed upstream.